crowbar_framework/lib/crowbar/lock.rb

```diff
@@ -0,0 +1,76 @@
+#
+# Copyright 2015, SUSE LINUX GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+module Crowbar
+  class Lock
+    attr_accessor :logger
+    attr_accessor :path
+    attr_accessor :remote
+
+    def initialize(options = {})
+      @logger = options.fetch :logger, Rails.logger
+      @path = options.fetch :path, "/var/lock"
+      @remote = options.fetch :remote, false
+    end
+
+    def create(name, nodes = [])
+      if @remote
+        remote(name, nodes, action: "create")
+      else
+        local(name, action: "create")
```

Comment:
So take this pull request as an option to start small refactoring, the number of places where it's used is pretty small:

```
crowbar/crowbar_framework/app/models/service_object.rb:146: FileLock.acquire(name, :logger => @logger)
crowbar/crowbar_framework/app/models/service_object.rb:150: FileLock.release(f, :logger => @logger)
crowbar/crowbar_framework/app/models/role_object.rb:289: role_lock = FileLock.acquire "role:#{@role.name}"
crowbar/crowbar_framework/app/models/role_object.rb:311: FileLock.release role_lock
crowbar/crowbar_framework/app/models/role_object.rb:319: role_lock = FileLock.acquire "role:#{@role.name}"
crowbar/crowbar_framework/app/models/role_object.rb:322: FileLock.release role_lock
crowbar/crowbar_framework/lib/file_lock.rb:18: class FileLock
crowbar/crowbar_framework/lib/crowbar/deployment_queue.rb:231: FileLock
```